### PR TITLE
Property Test for PoStProof

### PR DESCRIPTION
**Pull request by jdjaustin**
**Summary of changes**
Changes introduced in this pull request:
- Use quickcheck to test our custom implementations of `PoStProof`

**Reference issue to close (if applicable)**

Closes associated task in #1881 

**Other information and links**

N/A

### CODE REVIEW

File: types/src/sector/post.rs [hunk 1]
```diff
@@ -102,3 +104,62 @@ pub mod json {
...
+            $serializer($obj, &mut serde_json::ser::Serializer::new(&mut writer)).unwrap();
+            String::from_utf8(writer).unwrap()
+        }};
+    }
+
+    macro_rules! from_str_with {
+        ($str:expr, $deserializer:path) => {
+            $deserializer(&mut serde_json::de::Deserializer::from_str($str)).unwrap()
+        };
+    }
```
Comment (LesnyRumcajs) [inline types/src/sector/post.rs:157]: Same comment as [here](https://github.com/ChainSafe/forest/pull/1982#discussion_r986615383)

File: types/src/sector/post.rs [hunk 2]
```diff
@@ -102,3 +104,62 @@ pub mod json {
...
+                RegisteredPoStProof::StackedDRGWindow64GiBV1,
+            ])
+            .unwrap();
+        let postproof = PoStProof {
+            post_proof: *registered_postproof,
+            proof_bytes: Vec::arbitrary(g),
+        };
+        PoStProofWrapper { postproof }
+    }
+}
```
Comment (LesnyRumcajs) [inline types/src/sector/post.rs:137]: Could this be entirely inside `mod tests` along with the import?

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

**Review by lemmih — APPROVED** [review]

**Review by tyshko5 — APPROVED** [review]

File: types/Cargo.toml [hunk 3]
```diff
@@ -29,3 +29,9 @@ time                        = { version = "0.3", features = ["serde", "serde-wel
 
 [features]
 proofs = ["filecoin-proofs-api"]
+
+[dev-dependencies]
+forest_test_utils.workspace = true
+quickcheck                  = { version = "1" }
+quickcheck_macros           = "1"
+serde_json                  = "1.0"
```
Comment (LesnyRumcajs) [inline types/Cargo.toml:37]: use workspace dependencies?

**Review by LesnyRumcajs — APPROVED** [review]
LGTM, after you merge the changes from the PR introducing workspace dependencies for `quickcheck`, please update the `types/Cargo.toml` and potentially other `.toml` files that would be able to use it.

**Pull request merged**